datafusion/optimizer/src/filter_push_down.rs

```diff
@@ -336,6 +357,18 @@ fn optimize(plan: &LogicalPlan, mut state: State) -> Result<LogicalPlan> {
         LogicalPlan::Analyze { .. } => push_down(&state, plan),
         LogicalPlan::Filter(Filter { input, predicate }) => {
             let mut predicates = vec![];
+            let mut alias_cols_expr_and_name = HashMap::new();
+            //Need rewrite column name before push down
+            let input_plan = &*input.clone();
+            if let LogicalPlan::Projection(projection) = input_plan {
```

Comment:
I think i should get all `Expr::Alias`, not only one level🤔

Reply:
Is there one API to get all inputs'  all Expr::Alias

Reply:
I wonder if you can use the `LogicalPlan`'s output schema to detect the case you are looking for

I am still a little confused about where the issue is -- is the filter push down logic adding an extra alias? Or is it not adding an alias it should?